### DISCUSSION TITLE
Terraform Cloud integration - self-hosted option?

### DISCUSSION THREAD

**Discussion by ericofusco** [Checkmarx/kics#6008] [General] [answered]
Hi all, 👋 

I was looking at https://docs.kics.io/latest/integrations_tfcloud/ which suggests configuring the endpoint https://kics.io/tfc/event. 

I assume this runs on checkmarx infra with the latest kics version and default rules. Is that correct?

I wonder if this integration is open-source or the configuration to create the tfc endpoint is available somewhere. It would be great if we can easily self-host the integration and run kics with custom queries.

**Comment by cx-miguel-dasilva** [comment, marked as answer]
Hi @ericofusco, hope you are doing great!
The KICS TFCloud hook is a private implementation and does not support custom security queries. 
I am sorry for any inconvenience. Let us know in case you need something else 🙂.

**Reply by ericofusco** [reply]
Thanks @cxMiguelSilva. Appreciate the response.
